lib/maths/CBoostedTreeImpl.cc

```diff
@@ -542,8 +638,27 @@ void CBoostedTreeImpl::initializeTreeShap(const core::CDataFrame& frame) {
     }
 }
 
+void CBoostedTreeImpl::selectTreesToRetrain(const core::CDataFrame& frame) {
+
+    if (m_TreesToRetrain.empty() == false) {
+        return;
+    }
+
+    TDoubleVec probabilities{retrainTreeSelectionProbabilities(
```

Comment:
I pondered this as well. Basically, I wanted to find code I could move out of `CBoostedTreeImpl`: it is already rather large. It is also exposed for testing. We could, of course, have a different mechanism for testing internal details of the `CBoostedTreeImpl`. Since this change doesn't actually introduce the function and it'll be a largish change to refactor. I don't want to make it as part of this PR. I'll leave a TODO to investigate whether we move it back into `CBoostedTreeImpl`.

Reply:
See [5260aeb](https://github.com/elastic/ml-cpp/pull/1837/commits/5260aeb2a8e6b0d22c673a9844a1fb13aac145dd).